### PR TITLE
CSHARP-4732: Add SearchOptions overload to Search API

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver/AggregateFluent.cs [hunk 1]
```diff
@@ -244,11 +244,26 @@ public override IAggregateFluent<TNewResult> ReplaceWith<TNewResult>(AggregateEx
             SearchHighlightOptions<TResult> highlight = null,
             string indexName = null,
             SearchCountOptions count = null,
-            SortDefinition<TResult> sort = null,
```
Comment (BorisDog) [inline src/MongoDB.Driver/AggregateFluent.cs:247]: `Sort` parameter was added in [CSHARP-4728](https://jira.mongodb.org/browse/CSHARP-4728), and has not been released yet.

File: src/MongoDB.Driver/IAggregateFluent.cs [hunk 2]
```diff
@@ -361,25 +361,35 @@ public interface IAggregateFluent<TResult> : IAsyncCursorSource<TResult>
...
         /// Flag that specifies whether to perform a full document lookup on the backend database
         /// or return only stored source fields directly from Atlas Search.
         /// </param>
         /// <param name="scoreDetails">
         /// Flag that specifies whether to return a detailed breakdown
-        /// of the score for each document in the result. 
+        /// of the score for each document in the result.
         /// </param>
         /// <returns>The fluent aggregate interface.</returns>
         IAggregateFluent<TResult> Search(
```
Comment (sanych-sun) [inline src/MongoDB.Driver/IAggregateFluent.cs:373]: Do we want to mark the method as Obsolete and suggest to use newly created method with searchOptions parameter instead?

**Review by sanych-sun — APPROVED** [review]
LGTM + suggestion to mark "old" method as Obsolete to encourage new method usage.

File: src/MongoDB.Driver/IAggregateFluent.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Driver/IAggregateFluent.cs:373]: Discussed offline

**Pull request merged**